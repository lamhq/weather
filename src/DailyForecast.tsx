import React from 'react';
import styled from 'styled-components';

interface DailyForeCastProps {
  date: string;
  minTemp: number;
  maxTemp: number;
  weatherImgUrl: string;
}

const Container = styled.article`
  display: flex;
  flex-direction: column;
  align-items: center;
  height: 160px;
  margin-bottom: 30px;
`;

const Header = styled.header`
  font-size: 16px;
  font-weight: bold;
  margin-bottom: 10px;
`;

const WeatherImg = styled.img`
  max-width: 80px;
`;

const Footer = styled.footer`
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  width: 100%;
`;

const LowTemp = styled.div`
  color: #878787;
`;

const HiTemp = styled(LowTemp)`
  font-weight: bold;
`;

const DailyForeCast: React.FC<DailyForeCastProps> = ({ date, minTemp, maxTemp, weatherImgUrl }) => {
  return (
    <Container>
      <Header>{date}</Header>
      <WeatherImg src={weatherImgUrl} alt="" />
      <Footer>
        <LowTemp>{minTemp}&#xB0;</LowTemp>
        <HiTemp>{maxTemp}&#xB0;</HiTemp>
      </Footer>
    </Container>
  );
};

export default DailyForeCast;