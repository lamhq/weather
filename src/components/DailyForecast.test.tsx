import React from 'react';
import renderer from 'react-test-renderer';
import DailyForecast from './DailyForecast';

describe('DailyForecast component', () => {
  it('should match snapshot', () => {
    const tree = renderer
      .create(<DailyForecast date="Monday" minTemp={20} maxTemp={28} weatherImgUrl="/" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
