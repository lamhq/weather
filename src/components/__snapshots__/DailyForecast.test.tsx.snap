// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DailyForecast component should match snapshot 1`] = `
<article
  className="sc-AxjAm fmtPVS"
>
  <header
    className="sc-AxirZ jelnuR"
  >
    Monday
  </header>
  <img
    alt=""
    className="sc-AxiKw hFUAIY"
    src="/"
  />
  <footer
    className="sc-AxhCb cPNXDK"
  >
    <div
      className="sc-AxhUy jRsvKf"
    >
      20
      °
    </div>
    <div
      className="sc-AxhUy sc-AxgMl gtKIYK"
    >
      28
      °
    </div>
  </footer>
</article>
`;
